src/libraries/System.Net.Http/src/System/Net/Http/Headers/Int64NumberHeaderParser.cs

```diff
@@ -2,44 +2,58 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
 using System.Globalization;
 
 namespace System.Net.Http.Headers
 {
-    internal sealed class Int64NumberHeaderParser : BaseHeaderParser
+    internal sealed class Int64NumberHeaderParser : HttpHeaderParser
     {
         // Note that we don't need a custom comparer even though we have a value type that gets boxed (comparing two
         // equal boxed value types returns 'false' since the object instances used for boxing the two values are
         // different). The reason is that the comparer is only used by HttpHeaders when comparing values in a collection.
-        // Value types are never used in collections (in fact HttpHeaderValueCollection expects T to be a reference
-        // type).
+        // Value types are never used in collections (in fact HttpHeaderValueCollection expects T to be a reference type).
 
         internal static readonly Int64NumberHeaderParser Parser = new Int64NumberHeaderParser();
 
-        private Int64NumberHeaderParser()
-            : base(false)
+        private Int64NumberHeaderParser() : base(supportsMultipleValues: false)
         {
         }
 
         public override string ToString(object value)
         {
             Debug.Assert(value is long);
-
             return ((long)value).ToString(NumberFormatInfo.InvariantInfo);
         }
 
-        protected override int GetParsedValueLength(string value, int startIndex, object? storeValue,
-            out object? parsedValue)
+        public override bool TryParseValue(string? value, object? storeValue, ref int index, [NotNullWhen(true)] out object? parsedValue)
         {
-            ReadOnlySpan<char> span = value.AsSpan(startIndex).Trim();
-            if (HeaderUtilities.TryParseInt64(span, out long result))
+            ReadOnlySpan<char> span = value.AsSpan(index);
```

Comment:
You mean a helper that does
```C#
            span = span.Slice(HttpRuleParser.GetWhitespaceLength(span));
```
?

Reply:
I was thinking it would trim start and end, that way the parsing code could be something like:

```C#
            span = HttpRuleParser.TrimValidWhitespace(span);
            if (HeaderUtilities.TryParseInt64(span, out long result))
            {
                    parsedValue = result;
                    index = value!.Length;
                    return true;
                }
            }

            // Invalid.
            parsedValue = null;
            return false;
```

Reply:
Another possibility would be to introduce a base class like SingleValueHeaderParser or something like that, that does the necessary boilerplate for dealing with whitespace and adds a abstract method like

```C#
    protected abstract bool TryParseSingleValue(Span<char> trimmedSpan, out object? parsedValue);
```

Then the derived class code is minimal:

```c#
    protected override bool TryParseSingleValue(Span<char> trimmedSpan, out object? parsedValue)
    {
      if (HeaderUtilities.TryParseInt64(span, out long result)) 
      { 
        parsedValue = result; 
        return true; 
      }
    }
 
    parsedValue = null; 
    return false;
}
```

Reply:
> I was thinking it would trim start and end, that way the parsing code could be something like

We could... it's just a bit more expensive and cumbersome to walk backwards like that.

I'm just going to close this.  Was opportunistically improving something that jumped out at me while reviewing the code for another reason, but it's resulting in diving head first into the world of we don't really like how all of this header parsing is structured in general and not wanting to make these kinds of changes until the whole thing is just overhauled, whenever that may be.